src/main/java/com/epam/machine/controller/LoginServlet.java

```diff
@@ -24,14 +29,24 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
         String password = request.getParameter("password");
         response.setContentType("text/html");
         try {
```

Comment:
```java
@Override
    public void doPost(HttpServletRequest request, HttpServletResponse response) {
        String username = request.getParameter("username");
        String password = request.getParameter("password");
        session = request.getSession();
        try {
            if (clientService.checkCredentials(username, password)) {
                session.setAttribute("login", username);
                session.setAttribute("role", Roles.DRIVER);
                response.sendRedirect("/clients/profile");
            } else if (adminService.checkCredentials(username, password)) {
                session.setAttribute("login", username);
                session.setAttribute("role", Roles.ADMIN);
                response.sendRedirect("/clients/profile");
            } else {
                request.setAttribute("fail", 1);
                request.getRequestDispatcher("/login.jsp").forward(request, response);
            }
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }
```

Reply:
It's a little bit better, so u should also create methods in AdminService and ClientService to check credentials and enum for Roles in our project.
Service should call method from its own repository if it returns null -> false otherwise true